include/RED4ext/Buffer-inl.hpp

```diff
@@ -4,6 +4,32 @@
 #include <RED4ext/Buffer.hpp>
 #endif
 
+RED4EXT_INLINE RED4ext::RawBufferAllocator::RawBufferAllocator(Memory::IAllocator* aAllocator)
+    : allocator(aAllocator ? *reinterpret_cast<uintptr_t*>(aAllocator) : 0)
+{
+}
+
+RED4EXT_INLINE void* RED4ext::RawBufferAllocator::ReallocAligned(void* aData, uint32_t aSize, uint32_t aAlignment) const
+{
+    auto innerAllocator = reinterpret_cast<const Memory::IAllocator*>(&allocator);
```

Comment:
I see that the allocator can be `0`. I think a check for this would be good.

Reply:
It should never be unset, but I'm unsure how to enforce it, so I'll add a check for now.

Reply:
We can throw if needed.